Bump version to 0.13.0

selliott512 opened the pull request:
This includes all places that the version appears as well as a new "NEWS" section with the new changes.

--

### Don't merge yet - synchronize with tagging

This is a work in progress. I'll update as we talk about how to summarize what happened in #development. The website will probably have to be updated at the same time as well.

selliott512 commented:
For news the categories, such as "Sprites" seem arbitrary. I'm quite open to suggestions.

selliott512 commented:
Hey @chungy , as far as I know we're very close to a release. We just need to add a few screen shots to the website (which I don't have write permission to, and which I have some questions about), but the main repository is super close (pending this PR, adding a few entries to "CREDITS" and rebuilding the nodes, all of which there are PRs for).

There's also a matter of signing the tags (which normally you do) and whatever one clicks in the GitHub GUI (and elsewhere?) to do the FD release.

mc776 commented:
> For news the categories, such as "Sprites" seem arbitrary. I'm quite open to suggestions.

"Sprites" seems perfectly natural to me, but that's from the perspective of a modder and contributor.

Right now the only two items listed in that category are (a) about monsters and (b) phrased in terms of the monster being the base unit. Maybe the categories could be "Monsters", "Weapons", "Music" and "Levels", with everything else that won't fit neatly in those being "Visuals", "Sounds" and "Misc"?

selliott512 commented:
Ok, I'll get back to this as the release gets closer. I'll also need to update the dates. If you, or anyone, wants to give me some text to address the suggestion I'll take a shot at it.

selliott512 commented:
In part to let everyone know this is something we're still trying to do I updated to latest master and touched all he dates. I don't know the the actual date will be, but I'll touch the dates again then.

I also took a shot at @mc776 's suggestions, but the result may not be optimal. If you know of things that should be showcased just let me know. See `NEWS.adoc`.

The new section in `NEWS.adoc`, from `General` to `Weapons`, will also be used for the 0.13.0 release text that can be seen by selecting a release in GitHub. Currently for 0.12.1 it looks like [this](https://github.com/freedoom/freedoom/releases/tag/v0.12.1).

Rather than spamming a bunch of git log here's how to get an overview of the changes since 0.12.1 from a Linux, or similar, command line for folks who want to look through them:

If you haven't already it's helpful to have an `upstream` remote in your fork so that you don't have to worry about `origin` being out-of-date:
```
git remote add upstream git@github.com:freedoom/freedoom.git
```
Make sure your new, and all, remotes are up-to-date:
```
git fetch --all
```
Determine when 0.12.1 was in upstream master:
```
git log -F --grep 'bump version to 0.12.1' upstream/master

commit e532a931d8e0441877c0340843625c42ec122e12 (tag: sle-0.12.1-master)
Author: Mike Swanson <mikeonthecomputer@gmail.com>
Date:   Tue Oct 22 11:27:59 2019 -0700

    bump version to 0.12.1
```
Search from 0.12.1 found above to now in upstream master:
```
git log --no-merges --oneline e532a931d8e0441877c0340843625c42ec122e12..upstream/master
```

selliott512 commented:
I bumped the version date again, updated `NEWS.adoc` based on #development feedback, and I updated the screenshot links in the `dist/io.github.freedoom.*.metainfo.xml` files.

mc776 commented:
Shouldn't the Graphics under HEAD be moved under 0.13.0 as well?

selliott512 commented:
I've just updated this:

- Rebased to lastest master.
- Changed release date to December 10th 2023 (see the #development chat).
- Updated `NEWS.adoc` based on [this issue](https://github.com/freedoom/freedoom/issues/1175).

selliott512 commented:
I made this a draft to avoid accidentally merging it before the release.

mc776 commented on NEWS.adoc:
```diff
@@ -6,6 +6,59 @@
     textures.  These are named after and in honor of Esa Repo (Espi).
   * A STARBR1 texture is now included as a counterpart to STARBR2.
```
This HEAD section can be removed now that its contents are reflected below.

selliott512 replied:
I agree. Deleted. It seems odd that we had an `HEAD` section when there's no reason that we wouldn't release any news worthy enough to be in the file.

HimoriK commented:
Release this already

selliott512 commented:
> Are we ready for a release?

Looks good as far as I know. Can we pick a date? @fragglet when is a good day for you since you play a critical role? December 15th? Do we need more time to look over things one last time?

I made [my PR](https://github.com/freedoom/freedoom/pull/1205) a draft. I'll probably close it soon and move it to my own repo.

mc776 commented on NEWS.adoc:
* Spanish and French manuals.

selliott512 replied:
This is now "French and Spanish translations." I figured it was in the "Manual" section anyway.

mc776 commented on NEWS.adoc:
As vain as I am about how the torches turned out, I think this line item should be removed since it's already implied in "Various revisions to sprites and textures."

selliott512 replied:
Thanks. Done. I was going to ask if you could look over `NEWS.adoc`.

I've also bumped the dates from `2023-12-10` to `2023-12-15`. Ill do it again if that's not the case.

mc776 commented:
So I think pretty much everything's ready to go, but of the people who've posted anything on the repository at all the past year(?) I believe @fragglet is the only person who has the credentials to actually post the release (and edit the website).

I believe @selliott512 and I are both of the opinion that the current state of the project is ready for release.

MerrittK commented:
I concur.
